### PR TITLE
Add deployment configs for jenkins-config-updater

### PR DESCRIPTION
**Pull request by paradigm**
/cc stevekuznetsov

### CODE REVIEW

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+metadata:
+  name: jenkins-config-updater-prow
+  annotations:
+    "openshift.io/display-name": jenkins-config-updater-prow
+  description: |
+    jenkins-config-updater manifests for prow
+parameters: []
+objects:
+- apiVersion: v1
+  kind: Route
```
Comment (kargakis) [inline cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: I don't think you need a route.

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  kind: Service
+  metadata:
+    name: jenkins-config-updater-prow
+  spec:
+    selector:
+      app: prow
+      component: jenkins-config-updater-prow
+    ports:
+    - port: 8888
+    type: NodePort
```
Comment (kargakis) [inline cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: Should be `ClusterIP`

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+    to:
+      kind: Service
+      name: jenkins-config-updater-prow
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: jenkins-config-updater-prow
```
Comment (kargakis) [inline cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: Can you parameterize NAME similarly to what we do in other templates?

File: cluster/ci/config/prow/jenkins-config-updater.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            rm -rf /go/src /go/pkg
+    strategy:
+      dockerStrategy:
+        from:
+          kind: DockerImage
+          name: openshift/origin-release:golang-1.8
+        imageOptimizationPolicy: SkipLayers
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/jenkins-config-updater.yaml]: @csrwng how can we base this buidlconfig off of `openshift/origin` and just add the `jenkins-config-updater` in? change the `git` source to a inline dockerfile that just copies?

Comment (csrwng) [reply cluster/ci/config/prow/jenkins-config-updater.yaml]: the Dockerfile can be inline or in the source repo ... doesn't matter.
It looks like it's already doing that: https://github.com/openshift/test-infra/pull/54/files#diff-250b79a38a6c04f921de7860c877896bR1
Just not based off of origin.

Comment (kargakis) [reply cluster/ci/config/prow/jenkins-config-updater.yaml]: Why do we need to base this off of `openshift/origin`?

File: cluster/ci/config/prow/jenkins-config-updater.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        uri: "${SOURCE_URL}"
+        ref: "${SOURCE_REF}"
+      type: Git
+      dockerfile: |
+        FROM openshift/origin-release:golang-1.8
+
+        COPY . /go/src/k8s.io/test-infra
+        RUN cd /go/src/k8s.io/test-infra && \
+            godep restore && \
+            CGO_ENABLED=0 GOOS=linux go install -a --ldflags '-extldflags "-static"' ./experiment/jenkins-config-updater && \
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply cluster/ci/config/prow/jenkins-config-updater.yaml]: We don't want to do that, do we? What do we gain by having one large buildconfig? These things are released independently -- also, without a separate BC right now we wouldn't be able to build the dev build of the `j-c-u` for testing

[the remark this answers is not in the document]
Comment (kargakis) [reply cluster/ci/config/prow/jenkins-config-updater.yaml]: Yeah, eventually we should have separate BCs for testing or updating single components assuming bug fixes (https://github.com/openshift/release/issues/302). Before that happens we need a way to ensure we can do bulk-updates though and I don't see that handled here.

[the remark this answers is not in the document]
Comment (kargakis) [reply cluster/ci/config/prow/jenkins-config-updater.yaml]: Opened https://github.com/openshift/release/pull/376.

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: For testing we did -- to not need to reconfigure `hook`

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: Again for testing

**Comment by stevekuznetsov** [comment]
Add to the makefile:

```makefile
jekins-config-updater-build:
    oc process -f cluster/ci/config/prow/jenkins-config-updater.yaml | oc apply -f -
.PHONY: jenkins-config-updater-build

jekins-config-updater:
    # $JENKINS_NAMESPACE is the namespace where Jenkins is deployed
    oc create serviceaccount jenkins-config-updater -o yaml --dry-run | oc apply -f -
    oc adm policy add-role-to-user edit system:serviceaccounts:$( oc project -q ):jenkins-config-updater -n "${JENKINS_NAMESPACE}"
    oc process -f cluster/ci/config/prow/openshift/jenkins-config-updater.yaml | oc apply -f -
.PHONY: jenkins-config-updater
```

File: cluster/ci/config/prow/openshift/jenkins-config-updater.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      component: jenkins-config-updater-prow
+  spec:
+    replicas: 1
+    revisionHistoryLimit: 2
+    template:
+      metadata:
+        labels:
+          app: prow
+          component: jenkins-config-updater-prow
+      spec:
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/openshift/jenkins-config-updater.yaml]: Set `deploymentconfig.spec.template.spec.serviceAccountName` to be `jenkins-config-updater`

File: Makefile [hunk 7]
```diff
@@ -127,3 +127,14 @@ jenkins-setup: jenkins
...
 .PHONY: jenkins-setup-dev
+
+jekins-config-updater-build:
+	oc process -f cluster/ci/config/prow/jenkins-config-updater.yaml | oc apply -f -
+.PHONY: jenkins-config-updater-build
+
+jekins-config-updater:
+	# $JENKINS_NAMESPACE is the namespace where Jenkins is deployed
+	oc create serviceaccount jenkins-config-updater -o yaml --dry-run | oc apply -f -
+	oc adm policy add-role-to-user edit system:serviceaccounts:$( oc project -q ):jenkins-config-updater -n "${JENKINS_NAMESPACE}"
```
Comment (stevekuznetsov) [inline Makefile]:
```
 oc adm policy add-role-to-user edit -z jenkins-config-updater
```

File: Makefile [hunk 8]
```diff
@@ -127,3 +127,14 @@ jenkins-setup: jenkins
 jenkins-setup-dev: jenkins
 	oc new-app -f jenkins/setup/jenkins-setup-template.yaml -p SKIP_PERMISSIONS_JOB=1 -p SOURCE_URL="${RELEASE_URL}" -p SOURCE_REF="${RELEASE_REF}"
 .PHONY: jenkins-setup-dev
+
+jekins-config-updater-build:
+	oc process -f cluster/ci/config/prow/jenkins-config-updater.yaml | oc apply -f -
+.PHONY: jenkins-config-updater-build
+
+jekins-config-updater:
+	# $JENKINS_NAMESPACE is the namespace where Jenkins is deployed
```
Comment (stevekuznetsov) [inline Makefile]: delete

**Comment by stevekuznetsov** [comment]
LGTM sans namespace flag changes

**Pull request merged**